app/pages/project/instances/instance/tabs/SettingsTab.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { format, formatDistanceToNow } from 'date-fns'
+import { useEffect, useId, useState, type ReactNode } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { apiQueryClient, useApiMutation, usePrefetchedApiQuery } from '~/api'
+import { ListboxField } from '~/components/form/fields/ListboxField'
+import { useInstanceSelector } from '~/hooks/use-params'
+import { addToast } from '~/stores/toast'
+import { Button } from '~/ui/lib/Button'
+import { FieldLabel } from '~/ui/lib/FieldLabel'
+import { LearnMore, SettingsGroup } from '~/ui/lib/SettingsGroup'
+import { TipIcon } from '~/ui/lib/TipIcon'
+import { links } from '~/util/links'
+
+Component.displayName = 'SettingsTab'
+export function Component() {
+  const instanceSelector = useInstanceSelector()
+
+  const [now, setNow] = useState(new Date())
+
+  useEffect(() => {
+    const timer = setInterval(() => setNow(new Date()), 1000)
+    return () => clearInterval(timer)
+  }, [])
+
+  const { data: instance } = usePrefetchedApiQuery('instanceView', {
+    path: { instance: instanceSelector.instance },
+    query: { project: instanceSelector.project },
+  })
+
+  const instanceUpdate = useApiMutation('instanceUpdate', {
+    onSuccess() {
+      apiQueryClient.invalidateQueries('instanceView')
+      addToast({ content: 'Instance auto-restart policy updated' })
+    },
+  })
+
+  const form = useForm({
+    defaultValues: {
+      autoRestartPolicy: instance.autoRestartPolicy,
+    },
+  })
+  const { isDirty } = form.formState
+
+  const onSubmit = form.handleSubmit((values) => {
+    instanceUpdate.mutate({
+      path: { instance: instanceSelector.instance },
+      query: { project: instanceSelector.project },
+      body: {
+        ncpus: instance.ncpus,
+        memory: instance.memory,
+        bootDisk: instance.bootDiskId,
+        autoRestartPolicy: values.autoRestartPolicy,
+      },
+    })
+  })
+
+  return (
+    <form className="space-y-6" onSubmit={onSubmit}>
+      <SettingsGroup.Container>
+        <SettingsGroup.Header>
+          <SettingsGroup.Title>Auto-restart</SettingsGroup.Title>
+          <p>The auto-restart policy configured for this instance</p>
+        </SettingsGroup.Header>
+        <SettingsGroup.Body>
+          <ListboxField
+            control={form.control}
+            name="autoRestartPolicy"
+            label="Policy"
+            description="If unconfigured this instance uses the default auto-restart policy, which may or may not allow it to be restarted."
+            placeholder="Default"
+            items={[
+              { value: '', label: 'None (Default)' },
```

Comment:
I have a slight preference to just say "Default" here, rather than "None (Default)". I don't want to give the incorrect impression that not setting a policy is equivalent to disabling auto-restart when currently, we always default to "Best Effort".

Reply:
Sounds good – I'd included `None` because it makes the API more intuitive to a user that has used the console, but default is clearer.

I suppose the argument for enforcing `auto_restart_policy` not null, and adding a `default` enum on the API is that it means the user is forced to include it at create time, when for most users they shouldn't need to touch this?